doc/source/workflows/events.rst

```diff
@@ -99,6 +100,111 @@ After the workflow finishes checkpointing the event, the event listener will be
+
+.. code-block:: python
+
+    import ray
+    from ray import workflow
+    from ray.workflow.http_event_provider import HTTPListener
+
+    # Create a task waiting for an http event with a JSON message.
+    # The JSON message is expected to have an event_key field
+    # and an event_payload field.
+    event_task = workflow.wait_for_event(HTTPListener, event_key='my_event_key')
+
+    key, messsage = workflow.run(event_task, workflow_id='workflow_receive_event_by_http')
+
+workflow_id as part of the request URL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, `HTTPEventProvider` is deployed as a Ray serve endpoint at
+`http://hostname:port/event/send_event/<workflow_id>`. If it is deployed locally,
+the endpoint may be reached at `http://127.0.0.1:8000/event/send_event/<workflow_id>`.
+Note that `workflow_id` must be included in the web request to match the active
+`wait_for_event` task.
+
+If there are more than one active `wait_for_event` tasks within the same
+`workflow_id`, `HTTPEventProvider` will forward to the task subscribing to
+the matching `event_key`.
+
+Expected event content in JSON
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, `HTTPEventProvider` expects an arriving message to be a parsable JSON
+document with two fields, `event_key` and `event_payload`, as shown in the example
+below. The `event_key` field must match the `event_key` argument of the `workflow.wait_for_event`
+task. The value of the `event_payload` field is returned, along with the value of
+the `event_key` field as the output of the task.
+
+.. code-block:: json
+
+    {"event_key": "my_event_key", "event_payload": "my_event_message"}
+
+Example event submitter client
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An external event submitter needs to have:
+
+* the HTTP endpoint address (e.g. `http://127.0.0.1:8000/event/send_event/`)
+* the `workflow_id` (e.g. "workflow_receive_event_by_http")
+* a valid JSON formatted message
+
+The code snippet below retries submission by checking the HTTP response code 404,
+which is returned by `HTTPEventProvider` if the `event_key` was not expected or
+the `workflow_id` did not exist, HTTP response code 200 is returned if the event
+was submitted and accepted successfully. HTTP response code 500 indicates server
+side event processing failure, which could be caused by workflow cancellation.
+
+.. code-block:: python
+
+    from time import sleep
+    import requests
+
+    def send_event():
+        resp = requests.post(
+            "http://127.0.0.1:8000/event/send_event/"
+            + "workflow_receive_event_by_http",
+            json={"event_key": "my_event_key", "event_payload": "my_event_message"},
+        )
+        return resp
+
+    # repeat send_event() until the returned status code is not 404
+    while True:
+        res = send_event()
+        if res.status_code == 404:
+            sleep(0.5)
+        else:
+            break
+
+HTTP event listener
```

Comment:
Not sure, but it probably not necessary for this doc? The audience of this doc should be the user who want to use this. Not sure whether it's helpful for them.

Reply:
Removed